### PR TITLE
[Cleanup] Untangled the 'Streamable' packet queue manipulation

### PR DESCRIPTION
**Pull request by ulteq**

### CODE REVIEW

**Comment by only-a-ptr** [comment]
Kudos for fightting the `Stream*` monster. Cleanups are good. But I have some notes.

File: source/main/network/Streamable.cpp [hunk 1]
```diff
@@ -39,6 +39,8 @@ Streamable::~Streamable()
 void Streamable::addPacket(int type, unsigned int len, char* content)
 {
 #ifdef USE_SOCKETW
+	std::lock_guard<std::mutex> lock(m_send_work_mutex);
```
Comment (only-a-ptr) [inline source/main/network/Streamable.cpp]: Ahem.... commit says **[Cleanup]**
Adding a new mutex is totally not a cleanup

File: source/main/physics/Beam.cpp [hunk 2]
```diff
@@ -4498,13 +4496,14 @@ void Beam::updateDebugOverlay()
 
 void Beam::updateNetworkInfo()
 {
+	if (!gEnv->network) return;
```
Comment (only-a-ptr) [inline source/main/physics/Beam.cpp]: No no _no_ NO! :angry: 
This is just so damn wrong. It's our codebase, our function, yet we're unable to call it right, so we check with a global state? Sorry but this is one of the worst kinds of bullshit RoR is plagued with. Worse yet, it comes in a **[cleanup]** commit ... seriously?

Comment (ulteq) [inline source/main/physics/Beam.cpp]: I did not add that line, I only moved it to the top of the function.

File: source/main/network/Streamable.cpp (hunk 1 above)
Comment (ulteq) [inline source/main/network/Streamable.cpp]: Better than having a race on the packet queue?

File: source/main/physics/Beam.cpp (hunk 2 above)
Comment (only-a-ptr) [inline source/main/physics/Beam.cpp]: Oh, sorry... I really should stop skimming thru the code from work :blush:

File: source/main/network/Streamable.cpp (hunk 1 above)
Comment (only-a-ptr) [inline source/main/network/Streamable.cpp]: > Better than having a race on the packet queue?

Absolutely, but that's a honorable [bugfix], not a cleanup.

File: source/main/physics/Beam.cpp (hunk 2 above)
Comment (ulteq) [inline source/main/physics/Beam.cpp]: No problem.

**Comment by only-a-ptr** [comment]
Now it's perfect.
Keep up the good work.

**Pull request merged**